File: dayflow-hrms/backend/src/main/java/com/dayflow/hrms/service/AttendanceService.java
package com.dayflow.hrms.service;

import com.dayflow.hrms.dto.attendance.*;
import com.dayflow.hrms.enums.AttendanceStatus;

import java.time.LocalDate;
import java.util.List;

public interface AttendanceService {
    AttendanceDto checkIn(Long userId, CheckInRequest request, String ipAddress);
    AttendanceDto checkOut(Long userId, CheckOutRequest request, String ipAddress);
    DailyAttendanceSummaryDto getTodaySummary(Long userId);
    List<AttendanceDto> getMyAttendanceHistory(Long userId);
    List<AttendanceDto> getMyCalendarAttendance(Long userId, int year, int month);
    List<AttendanceDto> getGlobalAttendance(Long userIdFilter, LocalDate dateFilter, String departmentFilter, AttendanceStatus statusFilter);
    AttendanceDto manualAttendanceOverride(AttendanceOverrideRequest request, Long adminUserId, String ipAddress);
}

File: dayflow-hrms/frontend/src/services/attendanceService.js
import api from './api';

export const attendanceService = {
  checkIn: (data) => api.post('/attendance/check-in', data || {}),
  checkOut: (data) => api.post('/attendance/check-out', data || {}),
  getTodaySummary: () => api.get('/attendance/me/today'),
  getMyAttendance: () => api.get('/attendance/me'),
  getMyCalendar: (year, month) => api.get('/attendance/me/calendar', { params: { year, month } }),
  getGlobalAttendance: (params) => api.get('/admin/attendance', { params }),
  manualOverride: (data) => api.post('/admin/attendance/override', data)
};
